bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/handler/EchoHandler.java

```diff
@@ -304,6 +306,21 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                     account.forceCheckData();
                 }
             }
+            // DnD command
+            if (channelId.equals(CHANNEL_DND)) {
+                if (command == OnOffType.OFF) {
+                    connection.dnd(device, false);
+                    this.dnd = false;
+                    waitForUpdate = -1;
+                    account.forceCheckData();
+                }
+                if (command == OnOffType.ON) {
+                    connection.dnd(device, true);
+                    this.dnd = true;
+                    waitForUpdate = -1;
+                    account.forceCheckData();
+                }
```

Comment:
```suggestion
                boolean newDnd = OnOffType.ON.equals(command);
                connection.dnd(device, newDnd);
                this.dnd = newDnd;
                waitForUpdate = -1;
                account.forceCheckData();
```

Reply:
If I do this though, a REFRESH command command would set doNotDisturb to false, instead of doing nothing but refreshing it